spark/v2.4/spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -112,8 +122,10 @@ public DataWriterFactory<InternalRow> createWriterFactory() {
 
   @Override
   public void commit(WriterCommitMessage[] messages) {
-    if (replacePartitions) {
+    if (replacePartitions && overwriteFilter == null) {
       replacePartitions(messages);
+    } else if (replacePartitions) {
```

Comment:
Shouldn't this be `if (overwriteFilter != null)`? What happens if this isn't using overwrite mode but you give it an overwrite filter? Should it fail?

Reply:
It should, but the `overwriteFilter != null` condition is always true at this point, since otherwise it would entered the branch at line 125. Added the condition for better readability.

Good point about what happens when the overwrite mode is set to `append` but we give an overwrite filter. I throw an exception here when that happens. Should I throw here or rather on `IcebergSource` just before creating the writer?